commit 77ab6db0a1c403387b403e9351ab3f5ae1df83e6
Author: Jan Kiszka <jan.kiszka@web.de>
Date:   Mon Jul 14 12:28:51 2008 +0200

    KVM: VMX: Reinject real mode exception
    
    As we execute real mode guests in VM86 mode, exception have to be
    reinjected appropriately when the guest triggered them. For this purpose
    the patch adopts the real-mode injection pattern used in vmx_inject_irq
    to vmx_queue_exception, additionally taking care that the IP is set
    correctly for #BP exceptions. Furthermore it extends
    handle_rmode_exception to reinject all those exceptions that can be
    raised in real mode.
    
    This fixes the execution of himem.exe from FreeDOS and also makes its
    debug.com work properly.
    
    Note that guest debugging in real mode is broken now. This has to be
    fixed by the scheduled debugging infrastructure rework (will be done
    once base patches for QEMU have been accepted).
    
    Signed-off-by: Jan Kiszka <jan.kiszka@web.de>
    Signed-off-by: Avi Kivity <avi@qumranet.com>

diff --git a/include/asm-x86/kvm_host.h b/include/asm-x86/kvm_host.h
index da3c1979b734..83afa10c77f5 100644
--- a/include/asm-x86/kvm_host.h
+++ b/include/asm-x86/kvm_host.h
@@ -58,6 +58,9 @@
 
 #define DE_VECTOR 0
 #define DB_VECTOR 1
+#define BP_VECTOR 3
+#define OF_VECTOR 4
+#define BR_VECTOR 5
 #define UD_VECTOR 6
 #define NM_VECTOR 7
 #define DF_VECTOR 8
@@ -66,6 +69,7 @@
 #define SS_VECTOR 12
 #define GP_VECTOR 13
 #define PF_VECTOR 14
+#define MF_VECTOR 16
 #define MC_VECTOR 18
 
 #define SELECTOR_TI_MASK (1 << 2)